R/editing.R

```diff
@@ -979,3 +979,36 @@ update_package_object <- function(mn,
 
   return(pkg_new)
 }
+
+#' Helper for publish_object. Reformat the filName in system metadata.
+#'
+#' Reformat the fileName field in an object's system metadata to follow Arctic Data Center
+#' system metdata naming conventions.  Publish_object calls this function to rename
+#' the fileName field in system metadata.
+#'
+#' @param path (character) full file path
+#' @param sysmeta (S4) A system metadata object
+#'
+reformat_file_name <- function(path, sysmeta) {
+  base_name <- basename(path)
+  if (sysmeta@formatId == 'http://www.openarchives.org/ore/terms') {
+    ext <- '.rdf.xml'
+  } else if (grepl('eml://ecoinformatics\\.org/eml*', sysmeta@formatId)) {
+    ext <- '.xml'
+    # remove extension then truncate to 50 characters
+    base_name <- stringr::str_sub(base_name, end = -5) %>%
```

Comment:
you could use `tools::file_path_sans_ext` to more predictably remove just the extension here as opposed to the last 5 characters of the filename (though it really doesn't make a difference functionally)

Reply:
I like that change - definitely more elegant